src/cpufreq.c

```diff
@@ -82,9 +142,58 @@ static int cpufreq_read(void) {
     /* convert kHz to Hz */
     v.gauge *= 1000.0;
 
-    cpufreq_submit(i, v);
-  }
+    cpufreq_submit(i, "cpufreq", NULL, &v);
+
+    /* Read total transitions for cpu frequency */
+    if (report_total_trans) {
+      snprintf(filename, sizeof(filename),
+               "/sys/devices/system/cpu/cpu%d/cpufreq/stats/total_trans", i);
+      if (parse_value_file(filename, &v, DS_TYPE_COUNTER) != 0) {
+        WARNING("cpufreq plugin: Reading \"%s\" failed.", filename);
+        continue;
+      }
+      counter_t c = counter_diff(t_data[i].transition_prev, v.counter);
```

Comment:
IIRC, you haven't use `counter_diff` here, just do `cpufreq_submit` and all should work because `transitions` metric has `DERIVE` type.